newIDE/app/src/PropertiesEditor/index.js

```diff
@@ -225,9 +236,38 @@ export default class PropertiesEditor extends React.Component<Props, {||}> {
     );
   };
 
+  _renderDialog = (field: Field) => {
+    const EditorComponent = field.getDialog();
+
+    return (
+      <EditorComponent
+        key={field.name}
+        behavior={this.props.instances[0]}
+        project={this.props.project}
+        resourceSources={this.props.resourceSources}
+        onChooseResource={this.props.onChooseResource}
+        resourceExternalEditors={this.props.resourceExternalEditors}
+      />
+    );
+  };
+
   render() {
     const { mode } = this.props;
 
+    for (var i = 0, len = this.props.schema.length; i < len; i++) {
+      if (this.props.schema[i].getDialog) {
+        return (
```

Comment:
But how to make it to not render fields defined in the custom dialog?
For example if I render everything it will render my dialog and the `density` property, and I need the density property defined to use it in the dialog... or an I missing something?

Reply:
Let me think about it, but I think we can move toward a solution that is more similar to `ObjectsEditorService`, but for Behaviors (`BehaviorsEditorService`). i.e: we register the editor that you created for the behavior. If a behavior does not have a registered editor, then `BehaviorsEditorService` will create a PropertiesEditor, like what we do now.

I was thinking of doing the same for `ObjectsEditorService`: if you don't set an editor for your object, a default editor showing the properties will be displayed.
So both `ObjectsEditorService` and `BehaviorsEditorService` would work the same. It would avoid having a strange handling of "getDialog" in PropertiesEditor, allow custom editors while providing a default editor for most behaviors/objects.

Reply:
Nice!, will read the `ObjectsEditorService` in-depth.